### PR TITLE
refactor(flightsql): Use upstream `SqlInfoData` implementation

### PR DESCRIPTION
**Pull request by alamb**
This code has been ported upstream (see https://github.com/apache/arrow-rs/pull/4266) so we can use the upstream implementation rather than our copy in IOx

### CODE REVIEW

File: flightsql/src/sql_info/mod.rs [hunk 1]
```diff
@@ -133,164 +49,135 @@ static INSTANCE: Lazy<SqlInfoList> = Lazy::new(|| {
...
-        .with_sql_info(
-            SqlInfo::SqlSupportedConcurrenciesForResultSetScrollInsensitive,
-            0i32,
-        )
-        .with_sql_info(SqlInfo::SqlBatchUpdatesSupported, false)
-        .with_sql_info(SqlInfo::SqlSavepointsSupported, false)
-        .with_sql_info(SqlInfo::SqlNamedParametersSupported, false)
-        .with_sql_info(SqlInfo::SqlLocatorsUpdateCopy, false)
-        .with_sql_info(SqlInfo::SqlStoredFunctionsUsingCallSyntaxSupported, false)
+    let mut builder = SqlInfoDataBuilder::new();
```
Comment (alamb) [inline flightsql/src/sql_info/mod.rs:52]: There are no changes to the returned values, just the API changed slightly

File: flightsql/src/sql_info/mod.rs [hunk 2]
```diff
@@ -133,164 +49,135 @@ static INSTANCE: Lazy<SqlInfoList> = Lazy::new(|| {
...
+    builder.build().expect("Successfully built metadata")
 });
 
 /// Return a [`SqlInfoList`] that describes IOx's capablities
-pub fn iox_sql_info_list() -> &'static SqlInfoList {
+pub fn iox_sql_info_list() -> &'static SqlInfoData {
     &INSTANCE
 }
-
-#[cfg(test)]
```
Comment (alamb) [inline flightsql/src/sql_info/mod.rs:266]: these tests (and more) have been ported upstream. There is good end to end coverage

https://github.com/influxdata/influxdb_iox/blob/3d972561d5467531939fc9463a0a90fce7d38cbe/influxdb_iox/tests/end_to_end_cases/flightsql.rs#L1

**Review by appletreeisyellow — APPROVED** [review]
I like seeing those red blocks and removed file! Thank you for doing the cleanup @alamb!

**Review by crepererum — APPROVED** [review]

**Comment by alamb** [comment]
> I like seeing those red blocks and removed file! Thank you for doing the cleanup @alamb!

Yes, the red blocks give me a particular hit of dopamine

**Pull request merged**